apps/web/app/api/cancel/route.ts

```diff
@@ -26,6 +29,13 @@ async function handler(req: NextRequest) {
 
   const session = await getServerSession({ req: buildLegacyRequest(await headers(), await cookies()) });
 
+  // Rate limit: 10 booking cancellations per 60 seconds per user (or IP if not authenticated)
+  const identifier = session?.user?.id ? `cancel-${session.user.id}` : `cancel-ip-${piiHasher.hash(getIP(req))}`;
```

Comment:
Maybe as follows for more consistency.
_api:cancel-ip:ip_
_api:cancel-user: user_

I feel we need a fn to generate consistent identifiers given a request path.

Reply:
Yeah maybe worth making in a follow up - will fix these ones here though